### ISSUE TITLE
Connection is not returned to pool on exception

### ISSUE THREAD

**Issue by low205** [requery/requery#436]
When you are not using transactions and sql exception is thrown, connection are not released.

**Comment by npurushe** [comment]
Can you share what operation you called when the connection wasn't closed (or stack trace) ?

**Comment by low205** [comment]
Sure.

I tried with hikari on Oracle. Code fails with oracle's too many values exception, caused by inner sql select for user having too many field which can be fixed by specifying attribute, but connection is not returning to pool and it become exausted in no time. 

```
while (i++ < 30)
            try {
        data.count(Operation.class).where(OPERATION_USER.in(data
                        .select(User.class)
                        .where(User.LOGIN.upper().eq("LOGIN"))))
                        .get()
                        .value();
            } catch (Exception e) {
                 e.printStackTrace();
            }
```

We also had connection leaks on network connection losses. Requery is super cool, but with connection leaks like that it is hard to use in production...

**Comment by npurushe** [comment]
Thanks for the additional info, I've updated the behavior to close any connection/statement if there is an exception during the preparation/execution of the statement. Will have an update released soon with the change

**Comment by npurushe** [comment]
This fix is in version 1.2.0

**Comment by low205** [comment]
Thank you for fast responce!! Keep up the good work!
